### DISCUSSION TITLE
Vercel Create Project API Encoded response

### DISCUSSION THREAD

**Discussion by kashyapshiv** [vercel/community#1091] [Help] [answered]
Hello, I am calling the create Vercel project REST API from my custom API api/createVercel, but Its giving me encoded response and I am not able to get the repoId from it. 

But when I call the same API from Front-end component/file, its giving me proper response.

This wasn't the behaviour earlier, please ping if anyone has idea about this.

**Comment by swarnava** [comment, marked as answer]
Hi,

Can you share us some example such as endpoint details, Payload etc and how they are behaving differently?

**Reply by kashyapshiv** [reply]
Yes sorry for lack of information, 
Vercel Create Project API : https://api.vercel.com/v9/projects?teamId=xxx, 
Parameters: `
   {
              name: `${domainId}`,
              buildCommand: 'yarn build',
              environmentVariables: [
                {
                  key: 'KEY',
                  target: 'production',
                  value: 'VALUE',
                  type: 'plain',
                },
              ],
              framework: 'create-react-app',
              gitRepository: { repo: `REPO NAME`, type: 'github' },
              installCommand: 'yarn install',
            },
            { headers: { Authorization: 'AUTH TOKEN' } }
        `
Calling this API from server side (calling from api folder) gives encoded response 
![image](https://user-images.githubusercontent.com/105337972/204202366-f694cf23-2293-4698-b9fa-ab96e7b48046.png)

Also respond with status code 200 
![image](https://user-images.githubusercontent.com/105337972/204202670-dde25cdf-c3b3-4838-9236-2b4f394df6aa.png)

The same API when I call from Client side, It gives me proper response in data.

**Reply by amyegan** [reply]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or public repo that we can use? I was not able to reproduce this behavior in my test app.

**Reply by kashyapshiv** [reply]
Um, No I don't have public URL as of now, It was a weird issue, I switched to fetch instead of using axios for API calls and it works fine.

Don't know the cause of it but its working solution for now.

**Reply by amyegan** [reply]
I'm glad you got it working! It may have been an [`axios` encoding config problem](https://stackoverflow.com/questions/40211246/encoding-issue-with-axios). But [Next.js polyfills `fetch()`](https://nextjs.org/docs/basic-features/supported-browsers-features#server-side-polyfills) for your server code, so it's usually the best option anyway.

**Reply by kashyapshiv** [reply]
Thanks @amyegan :raised_hands:
